API Documentation

```diff
@@ -0,0 +1,740 @@
+# Chiquito API Documentation
+
+# Overall Workflow
+
+Chiquito is a DSL that compiles Chiquito AST to an IR which can be parsed by a Chiquito Halo2 backend and integrated into a Halo2 circuit. Therefore, to create a Halo2 circuit using Chiquito, we need to:
+
+- Call `circuit` function in DSL, which returns an `ast::Circuit` object. The `circuit` function defines the `ast::Circuit` object by:
+    - Importing Halo2 columns
+    - Generating fixed columns
+    - Adding forward signals
+    - Defining and instantiating “steps”, which defines internal signals, constraints, lookups, and witness generation
+- Create a `Compiler` object that compiles the `ast::Circuit` object to an `ir::Circuit` object
+- Call `chiquito2Halo2` function in Halo2 backend to convert the `ir::Circuit` object to a `ChiquitoHalo2` object
+- Integrate `ChiquitoHalo2` object into a Halo2 circuit using a boiler plate code
+
+# DSL Functions
+
+The above section describes the high level process of building and integrating a Chiquito Halo2 backend object into a Halo2 circuit. However, when building a circuit using Chiquito, the developer mostly call DSL functions to manipulate the `ast::Circuit` object, which the rest of this API documentation will focus on.
+
+Before we dive in, there are a few Chiquito-specific concepts that are not in Halo2:
+
+- Step (or super row): Chiquito provides syntax sugar for defining complex constraints and therefore sometimes requires allocating multiple Halo2 rows to build out a set of constraints. These groups of rows are called “super rows” or “steps”.
+- Signals: Halo2 uses rotations to describe constraints among different rows. Chiquito abstracts rotation away from developers by using “signals”, to mimic Circom’s user friendly features.
+    - Internal signal: only supports constraints within the same step
+    - Forward signal: supports constraints across different steps
```

Comment:
forward, also constraint on the value in the next step instance

Reply:
Edits
- Forward signal: supports constraints across different steps, including constraint on the value in the next step instance